MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -1,231 +1,152 @@
+#import <sqlite3.h>
+
 #import "MSDBStoragePrivate.h"
-#import "MSDatabaseConnection.h"
-#import "MSLogger.h"
 #import "MSMobileCenterInternal.h"
-#import "MSSqliteConnection.h"
-#import "MSUtility.h"
+#import "MSStorage.h"
+#import "MSUtility+File.h"
 
 @implementation MSDBStorage
 
-#pragma mark - Initialization
+- (instancetype)initWithSchema:(MSDBSchema *)schema filename:(NSString *)filename {
+  if ((self = [super init])) {
+    NSMutableArray *tableQueries = [NSMutableArray new];
+    NSMutableDictionary *dbColumnsIndexes = [NSMutableDictionary new];
+
+    // Path to the database.
+    NSURL *appSupportURL = [[[NSFileManager defaultManager] URLsForDirectory:NSApplicationSupportDirectory
+                                                                   inDomains:NSUserDomainMask] lastObject];
+    if (appSupportURL) {
+      NSURL *fileURL =
+          [appSupportURL URLByAppendingPathComponent:[kMSStorageDirectory stringByAppendingPathComponent:filename]];
+      [MSUtility createFileAtURL:fileURL];
+      _filePath = fileURL.absoluteString;
+    }
 
-- (instancetype)init {
-  self = [super init];
-  if (self) {
-    _connection = [[MSSqliteConnection alloc] initWithDatabaseFilename:kMSDBFileName];
-    _batches = [NSMutableDictionary<NSString *, NSArray<NSString *> *> new];
-    _capacity = NSUIntegerMax;
+    // Browse tables.
+    for (NSString *tableName in schema) {
+      NSMutableDictionary *tableColumnsIndexes = [NSMutableDictionary new];
+      NSMutableArray *columnQueries = [NSMutableArray new];
+      NSArray<NSDictionary *> *columns = schema[tableName];
+
+      // Optimization, don't even compute the query if the table already exists.
+      BOOL tableDontExist = ![self tableExists:tableName];
```

Comment:
I felt like maybe it doesn't worth building up the CREATE query which requires lots of string manipulation if the table already exists. Is it too much for almost nothing?